agent/discovery_chain_endpoint.go

```diff
@@ -0,0 +1,70 @@
+package agent
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+func (s *HTTPServer) ConnectDiscoveryChainGet(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	var args structs.DiscoveryChainRequest
+	if done := s.parse(resp, req, &args.Datacenter, &args.QueryOptions); done {
+		return nil, nil
+	}
+
+	args.Name = strings.TrimPrefix(req.URL.Path, "/v1/connect/discovery-chain/")
```

Comment:
🤔 Is `/connect` the right place for this?

For:
 - Discovery chain only works with Connect proxies now
 - We often use "Connect" as shorthand for "Service Mesh" which this is a feature of currently

Against:
 - The eventual scope of "discovery chain" might be as the canonical way to manage service discovery with or without proxies (features available differ)

I think I'd lean towards making `/discovery-chain` a new top-level API as it seems like it will over time become the new canonical way to manage both routing and discovery.

Reply:
Also what do you think about retrospectively adding a section to the L7 RFC for this to document the decision we come to?

Reply:
Moved to /v1/discovery-chain/<name>.

Will make a note to add to RFC afterwards.